pkg/kubernetes/run.go

```diff
@@ -13,24 +13,416 @@ limitations under the License.
+			app.CloseAppLogFile()
+			app.CloseDaprdLogFile()
+		} else {
+			logContext, cancel := context.WithCancel(context.Background())
+			rState.logCancel = cancel
+			err = setupLogs(logContext, app.AppID, daprdLogWriter, customAppLogWriter, podsInterface)
+			if err != nil {
+				print.StatusEvent(os.Stderr, print.LogWarning, "Error setting up logs for app %q present in %q . See logs directly from Kubernetes command line.: %s ", app.AppID, runFilePath, err.Error())
+			}
+		}
+
+		rState.deploymentFilePath = deploymentFilePath
+		rState.app = app
+
+		// append runSate only on successful k8s deploy.
+		runStates = append(runStates, rState)
+
+		print.InfoStatusEvent(os.Stdout, "Writing log files to directory : %s", app.GetLogsDir())
+	}
+
+	// If all apps have been started and there are no errors in starting the apps wait for signal from sigCh.
+	if !exitWithError {
+		print.InfoStatusEvent(os.Stdout, "Starting to monitor Kubernetes pods for deletion.")
+		go monitorK8sPods(monitoringContext, client, namespace, runStates, sigCh)
+		// After all apps started wait for sigCh.
+		<-sigCh
+		monitoringCancel()
+		print.InfoStatusEvent(os.Stdout, "Stopping Kubernetes pods monitoring.")
+		// To add a new line in Stdout.
+		fmt.Println()
+		print.InfoStatusEvent(os.Stdout, "Received signal to stop. Deleting K8s Dapr app deployments.")
+	}
+
+	closeErr := gracefullyShutdownK8sDeployment(runStates, client, namespace)
+	return exitWithError, closeErr
+}
+
+func createServiceConfig(app runfileconfig.App) serviceConfig {
+	return serviceConfig{
+		Kind:       serviceKind,
+		APIVersion: serviceAPIVersion,
+		Metadata: map[string]any{
+			nameKey: app.RunConfig.AppID,
+			labelsKey: map[string]string{
+				appLabelKey: app.AppID,
+			},
+		},
+		Spec: corev1.ServiceSpec{
+			Ports: []corev1.ServicePort{
+				{
+					Protocol:   tcpProtocol,
+					Port:       80,
+					TargetPort: intstr.FromInt(app.AppPort),
+				},
+			},
+			Selector: map[string]string{
+				appLabelKey: app.AppID,
+			},
+			Type: loadBalanceType,
+		},
+	}
+}
+
+func createDeploymentConfig(app runfileconfig.App) deploymentConfig {
+	replicas := int32(1)
+	dep := deploymentConfig{
+		Kind:       deploymentKind,
+		APIVersion: deploymentAPIVersion,
+		Metadata: map[string]any{
+			nameKey: app.AppID,
+		},
+	}
+
+	dep.Spec = appV1.DeploymentSpec{
+		Replicas: &replicas,
+		Selector: &metav1.LabelSelector{
+			MatchLabels: map[string]string{
+				appLabelKey: app.AppID,
+			},
+		},
+		Template: corev1.PodTemplateSpec{
+			ObjectMeta: metav1.ObjectMeta{
+				Labels: map[string]string{
+					appLabelKey: app.AppID,
+				},
+				Annotations: app.RunConfig.GetAnnotations(),
+			},
+			Spec: corev1.PodSpec{
+				Containers: []corev1.Container{
+					{
+						Name:            app.AppID,
+						Image:           app.ContainerImage,
+						Env:             getEnv(app),
+						ImagePullPolicy: corev1.PullAlways,
```

Comment:
@mukundansundar Can you this pull policy be configurable?

`Never` is more appropriate for locally built images.  (And `IfNotPresent` is the k8s default, except if latest.)